src/main/surfrdMod.F90

```diff
@@ -518,16 +539,120 @@ subroutine surfrd_cftformat( ncid, begg, endg, wt_cft, fert_cft, cftsize, natpft
     wt_nat_patch(begg:,natpft_lb:natpft_size-1+natpft_lb) = array2D(begg:,:)
     deallocate( array2D )
+ 
 
   end subroutine surfrd_cftformat
 
-!-----------------------------------------------------------------------
+  subroutine surfrd_wtfates( ncid, begg, endg )
+
+    !--------------------------------------------------------------------------
+    !     This routine evaluates the natural and crop functional
+    !     type fractions in the surface file and returns them to
+    !     a single, concatenated vector.  These weights
+    !     are only used for a satellite phenology run.
+    !     Note that FATES will actually allocate a different number of patches
+    !     and will use a mapping table to connect its own pft and cft
+    !     definitions to those it finds in the surface file.
+    !--------------------------------------------------------------------------
+    
+    ! !USES:
+    use clm_instur      , only : wt_nat_patch, wt_lunit
+    use clm_varpar      , only : cft_size, natpft_size
+    use landunit_varcon , only : istsoil, istcrop
+    
+    ! !ARGUMENTS:
+    implicit none
+    type(file_desc_t), intent(inout) :: ncid         ! netcdf id
+    integer          , intent(in)    :: begg, endg
+
+    !
+    ! !LOCAL VARIABLES:
+    logical  :: readvar                              ! is variable on dataset
+    real(r8),pointer :: array2d_pft(:,:)                 ! local array
+    real(r8),pointer :: array2d_cft(:,:)                 ! local array
+    integer :: g,p
+    integer :: cft_dimlen,natpft_dimlen,dimid
+    
+    character(len=32) :: subname = 'surfrd_fates'! subroutine name
+    
+    call ncd_inqdlen(ncid, dimid, cft_dimlen, 'cft')
+    call ncd_inqdlen(ncid, dimid, natpft_dimlen, 'natpft')
+
+    ! double check that cft_dimlen+natpft_dimlen = natpft_size
+    if((cft_dimlen+natpft_dimlen).ne.natpft_size)then
+       call endrun( msg=' ERROR: PCT+CFT dimlen does not match natpft_size when fates is on'//errMsg(sourcefile, __LINE__))
+    end if
+    
+    allocate( array2d_cft(begg:endg,1:cft_dimlen) )
+    allocate( array2d_pft(begg:endg,1:natpft_dimlen) )
+    
+    call ncd_io(ncid=ncid, varname='PCT_CFT', flag='read', data=array2d_cft, &
+         dim1name=grlnd, readvar=readvar)
+    if (.not. readvar) call endrun( msg=' ERROR: PCT_CFT NOT on surfdata file'//errMsg(sourcefile, __LINE__))
+    
+    call ncd_io(ncid=ncid, varname='PCT_NAT_PFT', flag='read', data=array2d_pft, &
+         dim1name=grlnd, readvar=readvar)
+    if (.not. readvar) call endrun( msg=' ERROR: PCT_NAT_PFT NOT on surfdata file'//errMsg(sourcefile, __LINE__))
+
+    ! In fates, all the weights in both the cft and pfts go into this array
+    ! It is only used by SP mode, and it can choose what PFTs to align with
+    
+    wt_nat_patch(begg:,0:natpft_dimlen-1) = array2d_pft(begg:,:)
+    wt_nat_patch(begg:,natpft_dimlen:natpft_dimlen+cft_dimlen-1) = array2d_cft(begg:,:)
+
+    if(.false.)then
```

Comment:
Ryan I also wasn't sure about what was going on here. This code is obviously not invoked, but is the plan for it to be a future option? If so it's OK to keep it around, but we should clearly comment what's up with it.

Reply:
I thought the first option would be essentially equivalent to the second option, but was safer regarding div0's. However, they don't give the same result. The second option, which is active, is necessary for b4b against the base.

Reply:
Ahhh, this is the non-bfb thing you found then. The differences you talked about were really subtle and it's surprising it came down to this part of the code. So something we could do when we are comfortable with answer changes is to activate the top part and remove the bottom part. I'm gathering that's what you think we would do at some point right?

Reply:
yes, that was my plan

Reply:
OK, as a future PR, I'll refactor to have the two options in two small subroutines that can be called.